pkg/controllers/static/deprovisioning/controller.go

```diff
@@ -81,8 +81,10 @@ func (c *Controller) Reconcile(ctx context.Context, np *v1.NodePool) (reconcile.
 		return reconcile.Result{}, nil
 	}
 
-	runningNodeClaims, _ := c.cluster.NodePoolState.GetNodeCount(np.Name)
+	runningNodeClaims, _, _ := c.cluster.NodePoolState.GetNodeCount(np.Name)
 	desiredReplicas := lo.FromPtr(np.Spec.Replicas)
+	// We dont wanna run into deprovisioning and drift controller races so we
+	// Only count running NodeClaims for deprovisioning as drifting NodeClaims will have replacements
```

Comment:
nit: I think it's a good idea to add a few more details on the race here, it took me a moment to think through the exact conditions which this could cause an issue. It would probably take a fair bit longer for someone unfamiliar.
```suggestion
	// We don't want to run into deprovisioning and disruption controller races so we
	// only count running NodeClaims for deprovisioning. If we counted both active and 
	// NodeClaims pending disruption here, we could burst above the desired replica count 
	// while the replacements are being created.
```